src/foam/flow/DocumentationFolderDAO.js

```diff
@@ -82,13 +69,29 @@ if ( ! id.matches("^[a-zA-Z0-9_-]+$") ) {
     {
       name: 'put_',
       javaCode: `
-verifyId(((String)getPK(obj)));
+String id = (String)getPK(obj);
+verifyId(id);
 
-java.nio.file.FileSystem fs = java.nio.file.FileSystems.getDefault();
+java.nio.file.Path path = new foam.nanos.fs.FileSystemStorage(getDir()).getPath(id + ".flow");
+if ( ! java.nio.file.Files.exists(path) ) {
```

Comment:
This could probably live inside the FileSystemStorage class, it could create the parent directory automatically if it doesn't exist, then client code doesn't need to know or care, they just ask for an outputStream as usual.